CustomGallery/src/main/java/com/likeminds/customgallery/utils/permissions/Permission.kt

```diff
@@ -49,13 +53,22 @@ class Permission private constructor(
         // returns the [PermissionExtras] for gallery permissions request
         @RequiresApi(Build.VERSION_CODES.TIRAMISU)
         fun getGalleryPermissionExtras(context: Context): PermissionExtras {
-            return PermissionExtras.Builder()
-                .permissions(
+            val permissionsArray =
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
+                    arrayOf(
+                        READ_MEDIA_IMAGES,
+                        READ_MEDIA_VIDEO,
+                        READ_MEDIA_VISUAL_USER_SELECTED
+                    )
+                } else {
```

Comment:
There should be a separate check for Android 14 as in Android 13 `READ_MEDIA_VISUAL_USER_SELECTED` permission was not present.